packages/node/test/integration/uninstall-concurrent-mixed.spec.ts

```diff
@@ -0,0 +1,143 @@
+import { NetworkContextForTestSuite } from "@counterfactual/local-ganache-server";
+import { One } from "ethers/constants";
+import { parseEther } from "ethers/utils";
+
+import { Node } from "../../src";
+import { CONVENTION_FOR_ETH_TOKEN_ADDRESS } from "../../src/constants";
+import { InstallMessage, NODE_EVENTS, ProposeMessage } from "../../src/types";
+import { toBeLt } from "../machine/integration/bignumber-jest-matcher";
+
+import { setup, SetupContext } from "./setup";
+import {
+  collateralizeChannel,
+  constructUninstallRpc,
+  constructUninstallVirtualRpc,
+  createChannel,
+  installVirtualApp,
+  makeInstallCall,
+  makeProposeCall
+} from "./utils";
+
+expect.extend({ toBeLt });
+
+jest.setTimeout(15000);
+
+// TODO: this test isnt working bc of the concurrency issues that are being investigated now.
+// reenable this test when it works
+describe.skip("Concurrently uninstalling virtual and regular applications without issue", () => {
+  let multisigAddressAB: string;
+  let multisigAddressBC: string;
+  let nodeA: Node;
+  let nodeB: Node;
+  let nodeC: Node;
+
+  beforeEach(async () => {
+    const context: SetupContext = await setup(global, true);
+    nodeA = context["A"].node;
+    nodeB = context["B"].node;
+    nodeC = context["C"].node;
+
+    multisigAddressAB = await createChannel(nodeA, nodeB);
+    multisigAddressBC = await createChannel(nodeB, nodeC);
+
+    await collateralizeChannel(
+      multisigAddressAB,
+      nodeA,
+      nodeB,
+      parseEther("2")
+    );
+
+    await collateralizeChannel(
+      multisigAddressBC,
+      nodeB,
+      nodeC,
+      parseEther("2")
+    );
+  });
+
+  it("can handle a virtual and regular TTT app uninstall", async done => {
+    let totalAppsUninstalled = 0;
+
+    // install a regular app
+    nodeB.on(NODE_EVENTS.PROPOSE_INSTALL, (msg: ProposeMessage) => {
+      makeInstallCall(nodeB, msg.data.appInstanceId);
+    });
+
+    const decodeNode = (pubId: string) => {
+      switch (pubId) {
+        case nodeA.publicIdentifier:
+          return "Node A";
+        case nodeB.publicIdentifier:
+          return "Node B";
+        case nodeC.publicIdentifier:
+          return "Node C";
+        default:
+          return "unknown";
+      }
+    };
+    nodeA.on(NODE_EVENTS.PROTOCOL_MESSAGE_EVENT, (msg: ProposeMessage) => {
+      console.log(
```

Comment:
we should probably remove all this logging before putting these tests into master